pwiz_tools/Skyline/SettingsUI/Irt/CalibrateIrtDlg.cs

```diff
@@ -662,87 +680,157 @@ public void SetPeptides(IEnumerable<MeasuredPeptide> peps)
+                foreach (var (pep, nodePep) in docPeptides)
+                {
+                    var allStats = doc.MeasuredResults.MSDataFileInfos
+                        .Select(info => resultsHandler.GetPeakFeatureStatistics(nodePep.Id.GlobalIndex, info.FileId.GlobalIndex))
+                        .Where(stats => stats != null).ToArray();
+                    var value = float.MaxValue;
+                    if (allStats.Length > 0)
                     {
-                        if (j + 1 > docPeptides.Count - 1 ||
-                                Math.Abs(docPeptides[j].RetentionTime - targetRT) <
-                                Math.Abs(docPeptides[j + 1].RetentionTime - targetRT))
+                        value = model is MProphetPeakScoringModel
+                            ? allStats.Select(stats => stats.QValue.Value).Max()
+                            : -allStats.Select(stats => stats.BestScore).Min();
+                    }
+                    if (!scoredPeptidesDict.TryGetValue(pep.Target, out var existing) || value < existing.Score)
+                        scoredPeptidesDict[pep.Target] = new ScoredPeptide(pep, nodePep, value);
+                }
+
+                var scoredPeptides = scoredPeptidesDict.Values.OrderBy(pep => pep.Peptide.RetentionTime).ToArray();
+                var minRt = scoredPeptides.First().Peptide.RetentionTime;
+                var maxRt = scoredPeptides.Last().Peptide.RetentionTime;
+                var rtRange = maxRt - minRt;
+
+                PeptideBucket[] buckets = null;
+                var bucketBoundaries = new[]
+                {
+                    minRt + rtRange * 1 / 8,
+                    minRt + rtRange * 2 / 8,
+                    minRt + rtRange * 4 / 8,
+                    minRt + rtRange * 6 / 8,
+                    minRt + rtRange * 7 / 8,
+                    double.MaxValue
+                };
+
+                // Try getting a regression using the known CiRT values
+                var cirtPeptidesAll = IrtStandard.CIRT.Peptides.ToDictionary(pep => pep.ModifiedTarget, pep => pep.Irt);
+                var scoredCirtPeptides = scoredPeptides.Where(pep => cirtPeptidesAll.ContainsKey(pep.Peptide.Target)).ToList();
+                var rts = scoredCirtPeptides.Select(pep => pep.Peptide.RetentionTime).ToList();
+                var irts = scoredCirtPeptides.Select(pep => cirtPeptidesAll[pep.Peptide.Target]).ToList();
+                var removedValues = new List<Tuple<double, double>>();
+
+                RegressionLine cirtRegression = null;
+                var cirtUsePredefined = false;
+
+                if (RCalcIrt.TryGetRegressionLine(rts, irts, peptideCount, out var regression, removedValues))
+                {
+                    for (var i = scoredCirtPeptides.Count - 1; i >= 0; i--)
+                    {
+                        if (removedValues.FirstOrDefault(tuple => tuple.Item1.Equals(rts[i]) && tuple.Item2.Equals(irts[i])) != null)

```

Comment:
It is simpler to do:
if (removedValues.Contains(Tuple.Create(rts[i], irts[i])))